Std/Tactic/NoMatch.lean

```diff
@@ -1,65 +0,0 @@
-/-
-Copyright (c) 2021 Mario Carneiro. All rights reserved.
-Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Mario Carneiro
--/
-import Std.Tactic.OpenPrivate
-import Lean.Elab.Match
-import Lean.Elab.ElabRules
-
-/-!
-This adds support for the alternative syntax `match x with.` instead of `nomatch x`. It is more
-powerful because it supports pattern matching on multiple discriminants, like regular `match`, and
-simply has no alternatives in the match.
-
-Along the same lines, `fun.` is a nullary pattern matching function; it is equivalent to
-`fun x y z => match x, y, z with.` where all variables are introduced in order to find an
-impossible pattern. The `match x with.` and `intro.` tactics do the same thing but in tactic mode.
--/
-namespace Std.Tactic
-open Lean Elab Term Parser.Term
-
-/--
-The syntax `match x with.` is a variant of `nomatch x` which supports pattern matching on multiple
-discriminants, like regular `match`, and simply has no alternatives in the match.
--/
-syntax:lead (name := noMatch) "match " matchDiscr,* " with" "." : term
```

Comment:
We can't remove this file, it needs to be a deprecation

Reply:
Okay, I've added this back. It felt like an extra hoop to jump through to add the warnings manually, if there's a better method for deprecating syntax I'm happy to try again.

Reply:
Well ideally there would be something to copy paste from, but we don't deprecate syntax very often, this is the first time it has happened in std. (Lean core likes to just move fast and break things here, but std has a deprecation policy.)